n3fit/src/evolven3fit/evolven3fit_API.py

```diff
@@ -0,0 +1,163 @@
+            data = yaml.safe_load(fp)
+        pdf_dict[replica] = data
+    return pdf_dict
+
+
+# Temporary solution. Then it will be loaded from the theory itself
+def construct_eko_for_fit(usr_path):
+    """
+    Construct the eko operator needed for evolution of fitted pdfs
+
+    Parameters
+    ----------
+        usr_path: pathlib.Path
+            path to the folder containing the fit  
+    Returns
+    -------
+        : eko.output.Output
+        eko operator
+        : dict
+        theory runcard
+        : dict
+        operator runcard
+    """
+    # read the runcard
+    my_runcard = utils.read_runcard(usr_path)
+    # theory_card construction
+    theory = Loader().check_theoryID(my_runcard["theory"]["theoryid"]).get_description()
+    theory.pop("FNS")
+    t_card = gen_theory.gen_theory_card(theory["PTO"], theory["Q0"], update=theory)
+    # construct operator card
+    op_x_grid = utils.generate_x_grid()
+    q2_grid = utils.generate_q2grid(theory["Q0"], 1.0e5)
+    op_card = gen_op.gen_op_card(q2_grid, update={"interpolation_xgrid": op_x_grid})
+    # generate eko operator (temporary because it will be loaded from theory)
+    eko_op = run_dglap(t_card, op_card)
+    return eko_op, t_card, op_card
+
+
+def evolve_exportgrid(exportgrid, eko, theory_card, operator_card):
+    """
+    Evolves the provided exportgrid for the desired replica with the eko and returns the evolved block
+
+    Parameters
+    ----------
+        exportgrid: dict
+            exportgrid of pdf at fitting scale
+        eko: eko object
+            eko operator for evolution
+        theory_card: dict
+            theory card
+        operator_card: dict
+            operator card
+    Returns
+    -------
+        : np.array
+        evolved block
+    """
+    # construct LhapdfLike object
+    pdf_grid = np.array(exportgrid["pdfgrid"]).transpose()
+    x_grid = np.array(exportgrid["xgrid"]).astype(np.float)
+    pdf_to_evolve = utils.LhapdfLike(pdf_grid, exportgrid["q20"], x_grid)
+    # evolve pdf
+    evolved_pdf = apply.apply_pdf(eko, pdf_to_evolve)
+    # generate block to dump
+    targetgrid = operator_card["interpolation_xgrid"]
+    block = genpdf.generate_block(
+        lambda pid, x, Q2, evolved_PDF=evolved_pdf: targetgrid[targetgrid.index(x)]
+        * evolved_PDF[Q2]["pdfs"][pid][targetgrid.index(x)],
+        xgrid=targetgrid,
+        Q2grid=operator_card["Q2grid"],
+        pids=br.flavor_basis_pids,
+    )
+    return block
+
+
+def dump_evolved_replica(evolved_block, usr_path, replica_num):
+    path_where_dump = usr_path / "nnfit" / usr_path.stem
+    # create folder to dump the evolved replica if it does not exist
+    if not os.path.exists(path_where_dump):
+        os.makedirs(path_where_dump)
+    to_write_in_head = "PdfType: replica\nFromMCReplica: " + str(replica_num) + "\n"
+    genpdf.export.dump_blocks(
+        path_where_dump, replica_num, [evolved_block], pdf_type=to_write_in_head
```

Comment:
To be improved inside `genpdf`: `evolved_block` has to be unraveled in the first two dimensions, maybe it should be done inside (documenting what you are expecting as input).

At the moment:
- a block has three dimensions `(Q2, x, flavor)`
- to call `genpdf.export.dump_blocks` you need to reduce them to two, i.e. `(Q2 * x, flavor)`, since you're asking for the table essentially (the one that will be dumped)

but it will make the difference if you reshape iterating faster on `x` or `Q2`, and this is specified by LHAPDF itself (I guess you have to iterate over Q2 as the very last thing).

Better to ask the user for something meaningful (e.g. the full three-dimensional object, with the dimensions sorted in a documented way), and then internally reshape. More user-friendly, no guesswork needed to pass the two-dimensional object :)